src/lib/server/websearch/markdown/utils/stringify.ts

```diff
@@ -0,0 +1,72 @@
+import type { SerializedHTMLElement } from "../../scrape/types";
+import { MarkdownElementType, type MarkdownElement } from "../types";
+
+// --- Markdown Elements ---
+
+/** Converts markdown element to a string with formatting */
+export function stringifyMarkdownElement(elem: MarkdownElement): string {
+	const content = elem.content.trim();
+	if (elem.type === MarkdownElementType.Header) return `${"#".repeat(elem.level)} ${content}\n\n`;
+	if (elem.type === MarkdownElementType.BlockQuote) {
+		return `${"> ".repeat(elem.depth)}${content}\n\n`;
+	}
+	if (elem.type === MarkdownElementType.CodeBlock) return `\`\`\`\n${content}\n\`\`\`\n\n`;
+
+	if (elem.type === MarkdownElementType.UnorderedListItem) return `- ${content}\n`;
+	if (elem.type === MarkdownElementType.OrderedListItem) {
+		const siblings = elem.parent?.children ?? [elem];
+		const currentIndex = siblings.indexOf(elem);
+		const lastAdjacentIndex = siblings
+			.slice(currentIndex + 1)
+			.findLastIndex((child) => child.type === MarkdownElementType.OrderedListItem);
+		const order = currentIndex - lastAdjacentIndex + 1;
+		return `${order}. ${content}\n`;
+	}
+
+	return `${content}\n\n`;
+}
+
+// ----- HTML Elements -----
+
+/** Ignores all non-inline tag types and grabs their text. Converts inline tags to markdown */
+export function stringifyHTMLElements(elem: (SerializedHTMLElement | string)[]): string {
```

Comment:
```suggestion
export function stringifyHTMLElements(elems: (SerializedHTMLElement | string)[]): string {
```

`elems` rather than `elem` would be a better argument name since it is an array